src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7992,6 +8029,25 @@ internal static bool IsReparsePoint(FileSystemInfo fileInfo)
                 : Platform.NonWindowsIsSymLink(fileInfo);
         }
 
+        internal static bool IsNameSurrogateReparsePoint(string filePath)
+        {
+#if !UNIX
+            var data = new WIN32_FIND_DATA();
+            using (SafeFileHandle handle = FindFirstFileEx(filePath, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0))
```

Comment:
Perhaps we could use method with fewer parameters
https://github.com/dotnet/corefx/blob/87089ef4559605f8a34ca8959a0ef9c71aa1e02e/src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs#L241

Reply:
CoreFx uses FindFirstFileEx() as well, they just name their [wrapper](https://github.com/dotnet/corefx/blob/09e2417cd0505df4558535651efb1bbcffdf0c59/src/Common/src/CoreLib/Interop/Windows/Kernel32/Interop.FindFirstFileEx.cs#L22) FindFirstFile()

Reply:
On the second point, I agree it might be an oversight that they don't have symmetric behavior.